### DISCUSSION TITLE
Basic fastlane screenshot and upload does not work as metadata about orientation not generated/used?

### DISCUSSION THREAD

**Discussion by pradeeproark** [fastlane/fastlane#27952] [General]
I have an ios App with ui tests to generate screenshots. The app only works in landscape orientation (at the moment). so I set the orientation in tests like so,

```
let app = XCUIApplication()
        XCUIDevice.shared.orientation = .landscapeLeft
        setupSnapshot(app)
        app.launch()
```

Fastfile

```
  lane :screenshots do
    capture_screenshots
    upload_to_app_store
  end
```

However I keep getting errors with image size rejection

`🚫 Error: ./fastlane/screenshots/en-GB/iPad mini (6th generation)-HomeScreen.png - Invalid screen size (Actual size is 2266x1488. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)`

This happens for all device types I specified in 

Snapfile

```
    "iPad (10th generation)",
    "iPad mini (6th generation)",
    "iPad Air 11-inch (M2)",
    "iPad Pro 13-inch (M4)",
```

It looks like when uploading them, app store api's are not informed that they are in landscape mode? I have tried

`frame_screenshots(white: true, force_orientation_block: "landscape_left")`

before upload_to_app_store. Still same results. Thoughts?

**Comment by bella964** [comment]
Hello, @pradeeproark 

The issue you're facing stems from the App Store Connect's requirement for screenshots to conform to specific dimensions for each device and orientation. When generating screenshots for landscape-only apps, ensuring the dimensions are correct and informing Fastlane about the landscape orientation is crucial.

Here’s how you can address the issue:

1. Verify Device Resolution and Orientation
Each device in landscape mode has specific screenshot resolution requirements. For instance:

iPad mini (6th generation): 1488 x 2266 (landscape is the same resolution, rotated).
iPad (10th generation): 1640 x 2360 (landscape mode is similarly rotated).
When generating screenshots in landscape mode, ensure your screenshots match these resolutions.

2. Adjust the Snapfile
Add explicit configuration for landscape orientation in your Snapfile. Fastlane uses the frame_screenshots function to generate device frames correctly, but it also relies on specifying the orientation in the Snapfile.

Update your Snapfile like this:

ruby
devices([
  "iPad (10th generation)",
  "iPad mini (6th generation)",
  "iPad Air (5th generation)",
  "iPad Pro (12.9-inch) (6th generation)"
])

languages([
  "en-GB"
])

force_orientation("landscape_left")
This ensures that the generated screenshots are interpreted as landscape.

3. Verify capture_screenshots in Fastfile
The capture_screenshots method generates screenshots based on the Snapfile configuration. Ensure that your app launches in the correct orientation during the UI test execution. Use:

ruby
lane :screenshots do
  capture_screenshots(
    force_rebuild: true,
    skip_open_summary: true
  )
  upload_to_app_store
end
4. Double-check the setupSnapshot Configuration
In your UI tests, you’ve correctly set the orientation with:

swift
let app = XCUIApplication()
XCUIDevice.shared.orientation = .landscapeLeft
setupSnapshot(app)
app.launch()
Make sure this snippet is included in every relevant test method generating screenshots. Fastlane relies on setupSnapshot to capture valid images.

5. Enable Debugging and Logging
To troubleshoot further, enable verbose logging during screenshot generation:

bash
fastlane screenshots --verbose
This provides detailed output and may highlight discrepancies in resolution or orientation.

6. Manually Verify Screenshots Before Upload
After generating screenshots, navigate to the folder where Fastlane stores them (e.g., ./fastlane/screenshots). Verify that the dimensions match the App Store Connect specifications for each device. If discrepancies persist, you might need to resize images using a script before uploading.
[safeco agent portal
](https://www.safe-coagent.com)
bash
mogrify -resize 1488x2266 ./fastlane/screenshots/en-GB/*.png
7. Fallback: Use upload_to_app_store with Flags
If the issue persists after adjusting dimensions, force the App Store upload to specify the orientation explicitly:

ruby
upload_to_app_store(
  skip_metadata: true,
  skip_app_version_update: true,
  force_upload: true
)


Best Regards
bella
